README.md

```diff
@@ -0,0 +1,203 @@
+
+  * `itemId`: The Wikidata id of the item being examined
+
+  * `resourceUrls`: List of objects describing external resources to scrape, and reference meta data to be added to references.
+
+    * `url`: String representing the url to scrape
+    * `referenceMetadata`: Data to append to the potential references matched at the end of this stage with optional reference keys. 
+
+    ***Note***: An example for this property can be found in the output provided by [SS4](#ss-4).
+    
+  * `statements`: List of unreferenced statements in the following structure:
+    
+    * `pid`: The property id of the statement
+    * `dataValue`: The data value of the statement / Item id of the Wikidata item value.
+  
+    Example:
+  
+      ```json
+    [
+        {
+            "pid": "P321", 
+            "dataValue": "2020-04-01",
+        },
+        {
+            "pid": "P777",
+            "dataValue": "Q666"
+        }
+        // ...
+    ]
+      ```
+
+## Pipe 3: Filter extracted values by plain text matching<a name="pipe-3"></a>
+
+This step will take the output of Pipe 2 as it's primary input. It will also take the output of SS2 as an input.
+
+One row of this file relates to one unreferenced statement. It contains the itemId the statement was on as well as all the extracted structured data that relates to this statement and referenceMetadata referring to where that data came from.
+
+The format of the template of one row of Pipe 2 is as follows:
+```json
+{
+  "statement": {statementBlob},
+  "itemId": [itemId],
+  "references": [ {
+    referenceMetadata: referenceBlob,
+    extractedData: [ [value] ]
+    } ]
+}
+```
+
+### referenceBlob format
+```json
+{
+  *[statedInPropId]: [externalIdItem],
+  *[externalIdProp]: [externalIdVal],
+  *[referenceUrl]: [String],
+  [dateRetrieved]: [String]
+}
+```
+
+### statementBlob format
+```json
+{
+  "propertyId": [String],
+  "datatype": [String],
+  "value": {valueBlob}
+}
+```
+
+### valueBlob
+See https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/Wikibase/+/master/docs/topics/json.md#Snaks-json_snaks for the datavalue -> value key.
+
+For example:
+```json
+{
+  "entity-type": "item",
+  "numeric-id": 350,
+  "id": "Q350"
+}
+```
+
+## Pipe 5: Format potential references into Quickstatements format<a name="pipe-5"></a>
+
+### input format
+```json
+{ "statement": {statementBlob}, "references": [ {referenceBlob}... ] }
+```
+
+## SS1: Find good external ID Properties<a name="ss-1"></a>
+
+A service to white-list external id properties based on a predefined blacklist, and the amount of Schema.org definitions found in a sample external resource.
+
+This service takes in a list of  manually blacklisted external ids, and retrieves a list of currently available external ids from Wikidata.
+
+The output for this service is a listed of string representation of white-listed external ids. For example:  
+
+  ```json
+  ['P1234', 'P1233', ...]
```

Comment:
This is not valid JSON.

Reply:
very true, I love that gh is now highlighting these! Will do :+1:

Reply:
Still `...` isn't valid JSON but I think leaving this improves clarity. The quotes are now fixed.